packages/edit-post/src/components/header/header-toolbar/index.js

```diff
@@ -25,24 +25,30 @@ function HeaderToolbar( { onToggleInserter, isInserterOpen } ) {
 		isInserterVisible,
 		isTextModeEnabled,
 		previewDeviceType,
-	} = useSelect(
-		( select ) => ( {
+	} = useSelect( ( select ) => {
+		const {
+			hasInserterItems,
+			getBlockRootClientId,
+			getBlockSelectionEnd,
+		} = select( 'core/block-editor' );
+		return {
 			hasFixedToolbar: select( 'core/edit-post' ).isFeatureActive(
 				'fixedToolbar'
 			),
 			// This setting (richEditingEnabled) should not live in the block editor's setting.
 			isInserterEnabled:
 				select( 'core/edit-post' ).getEditorMode() === 'visual' &&
 				select( 'core/editor' ).getEditorSettings().richEditingEnabled,
-			isInserterVisible: select( 'core/block-editor' ).hasInserterItems(),
+			isInserterVisible: hasInserterItems(
```

Comment:
Right now we can "disable" or "hide" the inserter, I wonder if we should just one technique in all cases (disable it) to avoid "jumps"

Reply:
So essentially you are suggesting a refactor that makes the inserter always visible, although sometimes it can be disabled?

Reply:
yes, something like that.